### PR TITLE
Adapt to pending Enzymecore changes

### PR DESCRIPTION
**Pull request by wsmoses**

### CODE REVIEW

File: ext/EnzymeExt.jl [hunk 1]
```diff
@@ -253,7 +255,7 @@ function gpu_rev(
 end
 
 function EnzymeRules.augmented_primal(
-        config::Config,
+        config::RevConfig,
         func::Const{<:Kernel},
```
Comment (vchuravy) [inline ext/EnzymeExt.jl:259]: Do we need those kinds of changes with an abstract Config type?

Comment (wsmoses) [reply ext/EnzymeExt.jl:259]: Forward mode rules changed fairly fundamentally, so we'll definitely still need these changes

**Comment by avik-pal** [comment]
Can we re-trigger the CI with Enzyme 0.13 released

**Comment by avik-pal** [comment]
https://github.com/JuliaGPU/KernelAbstractions.jl/blob/d9062a3ce98954450d568699640ec81365a21d01/test/Project.toml#L13 needs a bump

**Comment by wsmoses** [comment]
@maleadt okay this passes locally confirmed for me. This is probably the better place to bump first, if you are willing to give it the thumbs up

**Review by maleadt — APPROVED** [review]
Superficially LGTM.

**Comment by wsmoses** [comment]
Going to merge and hopefully check CUDA.jl success, then we can cut various releases if all is green

**Pull request merged**